dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/PageViewSerializer.java

```diff
@@ -54,14 +60,42 @@ public void serialize(final PageView pageView, final JsonGenerator jsonGenerator
 
         if (null != pageView.getUrlContent()) {
 
-            final DotContentletTransformer transformer = new DotTransformerBuilder().defaultOptions().content(pageView.getUrlContent()).build();
-            final Map<String, Object>   urlContentletMap = transformer.toMaps().stream().findFirst().orElse(Collections.EMPTY_MAP);
-            pageViewMap.put("urlContentMap", urlContentletMap);
+            this.createObjectMapUrlContent(pageView.getUrlContent(), pageViewMap);
         }
 
         return pageViewMap;
     }
 
+    protected void createObjectMapUrlContent(final Contentlet urlContent, final Map<String, Object> pageViewMap) {
+
+        final DotContentletTransformer transformer   = new DotTransformerBuilder().defaultOptions().content(urlContent).build();
+        final Map<String, Object>   urlContentletMap = transformer.toMaps().stream().findFirst().orElse(Collections.EMPTY_MAP);
+        this.createObjectMapKeyValue(urlContent, urlContent.getContentType(), urlContentletMap);
+
+        pageViewMap.put("urlContentMap", urlContentletMap);
+    }
+
+    /**
+     * Get all key value fields into the contentlet (if any) and put as an object into the contentMap (returns it)
+     * @param contentlet    {@link Contentlet}
+     * @param contentType   {@link ContentType}
+     * @param contentletMap {@link Map}
+     * @return Map
+     */
+    protected Map<String, Object> createObjectMapKeyValue(final Contentlet contentlet, final ContentType contentType, final Map<String, Object> contentletMap) {
```

Comment:
I think this code should be part of a more generic Serializer - `DotContentletTransformer`- any content can have a key/value field, not just Pages